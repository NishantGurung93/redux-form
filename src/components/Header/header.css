.header-wrapper {
  display: flex;
  flex-direction: row;
  width: 480px;
  justify-content: space-between;
  border: 1px solid;
}

.header-wrapper div {
  width: 160px;
  text-align: center;
  border: 1px solid;
}

.active {
  background-color: #81bde6;
}
