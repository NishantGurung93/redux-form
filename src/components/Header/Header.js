import { useRouteMatch } from "react-router-dom";
import './header.css';

const Header = () => {
  const userPage = useRouteMatch("/details");
  const privacyPage = useRouteMatch("/privacy");
  const confirmationPage = useRouteMatch("/confirmation");

  return (
    <div className="header-wrapper">
      <div className={userPage ? 'active' : ''}> User</div>
      <div className={privacyPage ? 'active' : ''}> Privacy</div>
      <div className={confirmationPage ? 'active' : ''}> Done</div>
    </div>
  )
};

export default Header;
