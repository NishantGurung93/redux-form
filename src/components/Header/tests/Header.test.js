import { render } from "@testing-library/react";
import Header from "../Header";

jest.mock('react-router-dom', () => ({
  useRouteMatch: jest.fn().mockReturnValue({
    isExact: true,
    params: {},
    path: "/details",
    url: "/details",
  }),
}));

const renderComponent = () => render(
  <Header />
);

describe('Header component', () => {
  beforeEach(() => {
    window.location = {
      href: "http://localhost:3000/details"
    }
  })

  it('should render the component', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
