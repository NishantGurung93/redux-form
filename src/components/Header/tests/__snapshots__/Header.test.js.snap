// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render the component 1`] = `
<div>
  <div
    class="header-wrapper"
  >
    <div
      class=""
    >
       User
    </div>
    <div
      class=""
    >
       Privacy
    </div>
    <div
      class=""
    >
       Done
    </div>
  </div>
</div>
`;
